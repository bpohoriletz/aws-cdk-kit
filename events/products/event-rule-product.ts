import * as events from 'aws-cdk-lib/aws-events';

export class EventRuleProduct implements IEventRuleProduct {
  eventPattern?: events.EventPattern | undefined;
  targets?: events.IRuleTarget[] | undefined;
}

interface IEventRuleProduct extends events.RuleProps {
  eventPattern?: events.EventPattern | undefined;
  targets?: events.IRuleTarget[] | undefined;
}

export interface IEventRuleBuilder {
  setEventPattern?(pattern: events.EventPattern | undefined): IEventRuleBuilder;
  // maybe have on director
  addTarget?(target: events.IRuleTarget | undefined): IEventRuleBuilder;
  getResult(): events.RuleProps;
}

export interface IEventRuleBuilderConstructor {
  new (): IEventRuleBuilder;
}
