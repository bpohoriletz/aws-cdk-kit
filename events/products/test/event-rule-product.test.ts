import { EventRuleProduct } from '../event-rule-product';

describe('new EventRuleProduct()', () => {
  it('has correct defaults', () => {
    expect(new EventRuleProduct()).toMatchSnapshot();
  });
});
