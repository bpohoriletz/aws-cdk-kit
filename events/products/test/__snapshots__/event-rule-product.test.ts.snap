// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new EventRuleProduct() has correct defaults 1`] = `EventRuleProduct {}`;
